src/com/oltpbenchmark/api/collectors/PostgresCollector.java

```diff
@@ -1,84 +1,100 @@
- *                                                                            *
- *    http://www.apache.org/licenses/LICENSE-2.0                              *
- *                                                                            *
- *  Unless required by applicable law or agreed to in writing, software       *
- *  distributed under the License is distributed on an "AS IS" BASIS,         *
- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  *
- *  See the License for the specific language governing permissions and       *
- *  limitations under the License.                                            *
- ******************************************************************************/
+/*
+ * Copyright 2020 by OLTPBenchmark Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
 
 package com.oltpbenchmark.api.collectors;
 
-import java.sql.Connection;
-import java.sql.SQLException;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import com.oltpbenchmark.util.JSONUtil;
 
-import org.apache.log4j.Logger;
+import java.sql.*;
+import java.util.*;
 
 public class PostgresCollector extends DBCollector {
 
-    private static final Logger LOG = Logger.getLogger(PostgresCollector.class);
+    private static final String VERSION_SQL = "SELECT version();";
 
-    private static final String PARAMETERS_SQL = "SHOW ALL";
+    private static final String PARAMETERS_SQL = "SHOW ALL;";
 
-    private static final String METRICS_SQL = "SELECT * FROM %s";
-
-    private static final String[] METRICS_VIEWS = {
-            "pg_stat_archiver",
-            "pg_stat_bgwriter",
-            "pg_stat_database",
-            "pg_stat_database_conflicts",
-            "pg_stat_user_tables",
-            "pg_statio_user_tables",
-            "pg_stat_user_indexes",
-            "pg_statio_user_indexes"
+    private static final String[] PG_STAT_VIEWS = {
+            "pg_stat_archiver", "pg_stat_bgwriter", "pg_stat_database",
+            "pg_stat_database_conflicts", "pg_stat_user_tables", "pg_statio_user_tables",
+            "pg_stat_user_indexes", "pg_statio_user_indexes"
     };
 
-    public PostgresCollector(String dbUrl, String dbUsername, String dbPassword) {
-        super(dbUrl, dbUsername, dbPassword);
+    private final Map<String, List<Map<String, String>>> pgMetrics = new HashMap<>();
+
+    public PostgresCollector(String oriDBUrl, String username, String password) {
+
```

Comment:
In general, more logic gets pushed down here.. I'm not necessarily opposed, I doubt much reuse is achieved